### PR TITLE
chore : structure the layout of the project

### PR DESCRIPTION
**Pull request by Kareem24**
### Test Plan

1. test
- Click on all navlinks make sure it route to the expected element
- run the lint script and make sure no linter error
- run the format script to make sure the code is well formated
- change the route to each specifed path and make sure its displaying the right element on the screen

. how to test the app
- click on each click on each navlinks to take you to the respective route
- input boxes and buttons are yet not functional

### CODE REVIEW

**Review by adejam — CHANGES_REQUESTED** [review]
You code formatting can be better...please add prettier and eslint. Use these tools to format the code better...

Create a `.github/workflows/linters.yml` file and add this code to it.

This creates a github action that ensure you have eslint running before pushing 

```yml
name: Linters

on: pull_request

env:
  FORCE_COLOR: 1

jobs:
  eslint:
    name: ESLint
    runs-on: ubuntu-18.04
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-node@v1
        with:
          node-version: '12.x'
      - name: Setup ESLint
        run: |
          npm install
      - name: ESLint Report
        run: npx eslint .
```

File: src/store.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+const store = createStore(reducer);
+function deposit(amount) {
+  return { type: "account/deposit", payload: amount };
+}
+
+store.dispatch(deposit(200));
+console.log("store");
+console.log(store.getState());
```
Comment (adejam) [inline src/store.js]: Let's remove all console statement

File: src/store.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { createStore } from "@reduxjs/toolkit";
+const initialState = {
+  amount: 0,
+  loan: 0,
+  purpose: "",
```
Comment (adejam) [inline src/store.js]: what are these states for... are they also related to the book library project

File: src/pages/Categories.jsx [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const Categories = () => {
+  return (
+    <div className="p-4">
+      <button className="border-2 border-black px-2 py-1">check status</button>
+    </div>
+  );
+};
```
Comment (adejam) [inline src/pages/Categories.jsx]: This can be improved by doing a direct return

File: src/main.jsx [hunk 4]
```diff
@@ -2,6 +2,7 @@ import React from "react";
 import ReactDOM from "react-dom/client";
 import App from "./App.jsx";
 import "./index.css";
+import "./store";
```
Comment (adejam) [inline src/main.jsx]: Why do we need this store import here?

File: src/components/Navbar.jsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    <ul className="flex gap-4 p-4">
+      <li className="capitalize">
+        <Link to={"/"}>home</Link>
+      </li>
+      <li className="capitalize">
+        <Link to={"/categories"}>categories</Link>
+      </li>
+    </ul>
+  );
+};
```
Comment (adejam) [inline src/components/Navbar.jsx]: A direct return can be applied here

File: src/components/Form.jsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
+const Form = () => {
+  return (
+    <form className="flex flex-col gap-5 items-start justify-start">
```
Comment (adejam) [inline src/components/Form.jsx]: Direct return

File: src/components/Book.jsx [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+    <li className="flex w-full md:w-1/2 justify-between items-center gap-4 list-none my-4">
+      <p>
+        <span>{title}</span> by <span>{author}</span>
+      </p>
+      <button className="border-2 border-black px-2 py-1">delete</button>
+    </li>
+  );
+}
+Book.propTypes = {
+  book: PropTypes.object,
```
Comment (adejam) [inline src/components/Book.jsx]: This proptypes is not very strict and efficient... A `book` is an object, however, the properties are not specified which means any property can be passed into it

File: src/App.jsx [hunk 8]
```diff
@@ -1,16 +1,20 @@
...
-		<>
-			<h1 className="text-red-600 capitalize">books library</h1>
-		</>
-	);
+  return (
+    <>
+      <BrowserRouter>
+        <Navbar />
+        <Routes>
+          <Route path="/" element={<Books />} />
```
Comment (adejam) [inline src/App.jsx]: Seems our routes is not specifying an `index` routes. please look into this

**Review by adejam — CHANGES_REQUESTED** [review]
## Nice job

Just a few more changes to be made

File: src/main.jsx (hunk 4 above)
Comment (Kareem24) [reply src/main.jsx]: Would be removed... Used for testing if Redux is successfully installed

File: src/App.jsx (hunk 8 above)
Comment (Kareem24) [reply src/App.jsx]: All changes made as required

File: src/store.js (hunk 1 above)
Comment (Kareem24) [reply src/store.js]: All console.log removed

File: src/store.js (hunk 2 above)
Comment (Kareem24) [reply src/store.js]: Not related to the book library,
It's a practice code to text if Redux installation was successful

File: src/pages/Categories.jsx (hunk 3 above)
Comment (Kareem24) [reply src/pages/Categories.jsx]: Delete the return keywords and wrap jsx with bracket so as to use the arrow function direct return

File: src/components/Navbar.jsx (hunk 5 above)
Comment (Kareem24) [reply src/components/Navbar.jsx]: Done as above

File: src/components/Book.jsx (hunk 7 above)
Comment (Kareem24) [reply src/components/Book.jsx]: Done,
Make use of Prototypes.shape function to specify the value expected on each key

**Review by adejam — APPROVED** [review]

**Pull request merged**